### DISCUSSION TITLE
How to perform inference with a model I trained with docTR?

### DISCUSSION THREAD

**Discussion by JMaximilian** [mindee/doctr#716] [Q&A] [answered]
I have trained a recognition model on my dataset following your article on [VIN ](https://blog.mindee.com/vin-extraction-with-doctr/)
But the ocr_predictor is not recognizing any text in the images and returns empty results. 

Steps I took:
1. Trained the recognition model
`%run doctr/references/recognition/train_tensorflow.py "My_path_to\train" "My_path_to\val" "sar_resnet31" --vocab legacy_french --pretrained --wb --epochs 10 -b 16`

2. Loaded the Model
```
from doctr.io import DocumentFile
from doctr.models import ocr_predictor`

model = ocr_predictor(reco_arch='sar_resnet31',pretrained=True)
REC_CKPT = r"my_path_to\sar_resnet31_20211214-204139\weights"
model.reco_predictor.model.load_weights(REC_CKPT)
```

3. Inference
```
# Image loading
doc = DocumentFile.from_images(img_path)
# Models inference
result = model(doc)
```

The result returns an empty block for any image I feed: 
```
Document(
  (pages): [Page(
    dimensions=(69, 333)
    (blocks): []
  )]
)
```

Environment: 
Windows 10 Home
Python 3.7 
Jupyter Notebook

**Comment by fg-mindee** [comment, marked as answer]
Hello @JMaximilian :wave: 

Let's try to get you out of this problem!

1. At the end of your training, what were the validation metrics?
2. When loading your checkpoint, I'd recommend first loading it into a `recognition_predictor` to check that everything is working well. In your snippet you created an ocr_predictor = det_predictor + reco_predictor, but it looks like you're only interested in the last part
3. Considering the size of your "page" (69, 333), I assume your inputs only have a single character sequence (no multiple lines). If so, no need to go for the ocr_predictor, you only need the reco_predictor as suggested above :+1: 

FYI: on very small images, the current pretrained version of our text detection models is performing badly (cf. https://github.com/mindee/doctr/issues/639), that's why the end-to-end pipeline is not working well in your case under my previous assumptions!

Hope this helps!

**Reply by JMaximilian** [reply]
Thanks! That helped.

**Comment by khanfarhan10** [comment]
For me what worked is `load_state_dict` instead of `load_weights` :

```python
model.det_predictor.model.load_state_dict(torch.load(REC_CKPT))
```

**Reply by felixdittrich92** [reply]
Hi @khanfarhan10 👋 
Tensorflow: load_weights
Pytorch: load_state_dict
